CMake/RealmCore.cmake

```diff
@@ -127,81 +163,117 @@ function(download_realm_core core_version)
 
     set_property(TARGET realm PROPERTY INTERFACE_LINK_LIBRARIES Threads::Threads ${CRYPTO_LIBRARIES})
 
-    set(REALM_CORE_INCLUDE_DIR ${core_directory}/include PARENT_SCOPE)
-endfunction()
+    set(REALM_CORE_INCLUDE_DIR ${core_directory}/${include_directory} PARENT_SCOPE)
+endmacro()
+
+function(download_realm_core core_version)
+    if(APPLE)
+        set(basename "realm-core")
+        set(compression "xz")
+        set(platform "")
+    elseif(REALM_PLATFORM STREQUAL "Android")
+        set(basename "realm-core-android")
+        set(compression "gz")
+        set(platform "-android-x86_64")
+    endif()
+    set(tarball_name "${basename}-${core_version}.tar.${compression}")
+    set(url "https://static.realm.io/downloads/core/${tarball_name}")
+    set(temp_tarball "/tmp/${tarball_name}")
+    set(core_directory_parent "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}")
+    set(core_directory "${core_directory_parent}/realm-core-${core_version}")
+    set(tarball "${core_directory_parent}/${tarball_name}")
 
-macro(define_built_realm_core_target core_directory)
-    set(core_library_debug ${core_directory}/src/realm/librealm-dbg.a)
-    set(core_library_release ${core_directory}/src/realm/librealm.a)
+    set(core_library_debug ${core_directory}/librealm${platform}-dbg.a)
+    set(core_library_release ${core_directory}/librealm${platform}.a)
     set(core_libraries ${core_library_debug} ${core_library_release})
 
-    ExternalProject_Add_Step(realm-core ensure-libraries
-        COMMAND ${CMAKE_COMMAND} -E touch_nocreate ${core_libraries}
-        OUTPUT ${core_libraries}
-        DEPENDEES build
-        )
+    download_realm_tarball(${url} ${core_directory} "${core_libraries}")
+    define_realm_core_target(YES ${core_directory})
+endfunction()
 
-    add_library(realm STATIC IMPORTED)
-    add_dependencies(realm realm-core)
+function(download_realm_sync sync_version)
+    if(APPLE)
+        set(basename "realm-sync-cocoa")
+        set(compression "xz")
+        set(platform "")
+    elseif(REALM_PLATFORM STREQUAL "Android")
+        set(basename "realm-sync-android")
+        set(compression "gz")
+        set(platform "-android-x86_64")
```

Comment:
It might be better to use `x86` instead of `x86_64`. Just if we wish to run the code on an emulator.